libraries/apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/introspection/IntrospectionSchema.kt

```diff
@@ -1,87 +1,94 @@
 package com.apollographql.apollo3.ast.introspection
 
-import com.squareup.moshi.JsonClass
-import com.squareup.moshi.JsonReader
-import com.squareup.moshi.Moshi
-import dev.zacsweers.moshix.sealed.annotations.TypeLabel
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.okio.decodeFromBufferedSource
+import kotlinx.serialization.json.okio.encodeToBufferedSink
 import okio.Buffer
 import okio.BufferedSource
 import okio.ByteString.Companion.decodeHex
 import okio.buffer
+import okio.sink
 import okio.source
 import java.io.File
 
-@JsonClass(generateAdapter = true)
+@Serializable
+private data class IntrospectionSchemaEnvelope(
+    val data: IntrospectionSchema?,
+    val __schema: IntrospectionSchema.Schema?,
+)
+
+@Serializable
 data class IntrospectionSchema(
     val __schema: Schema,
 ) {
-  @JsonClass(generateAdapter = true)
+  @Serializable
   data class Schema(
       val queryType: QueryType,
       val mutationType: MutationType?,
       val subscriptionType: SubscriptionType?,
       val types: List<Type>,
   ) {
-    @JsonClass(generateAdapter = true)
+    @Serializable
     data class QueryType(val name: String)
 
-    @JsonClass(generateAdapter = true)
+    @Serializable
     data class MutationType(val name: String)
 
-    @JsonClass(generateAdapter = true)
+    @Serializable
     data class SubscriptionType(val name: String)
 
-    @JsonClass(generateAdapter = true, generator = "sealed:kind")
+    @Serializable
     sealed class Type {
       abstract val name: String
       abstract val description: String?
 
-      @TypeLabel("SCALAR")
-      @JsonClass(generateAdapter = true)
+      @SerialName("SCALAR")
+      @Serializable
       data class Scalar(
           override val name: String,
           override val description: String?,
       ) : Type()
 
-      @TypeLabel("OBJECT")
-      @JsonClass(generateAdapter = true)
+      @SerialName("OBJECT")
+      @Serializable
       data class Object(
           override val name: String,
           override val description: String?,
           val fields: List<Field>?,
           val interfaces: List<Interface>?,
       ) : Type()
 
-      @TypeLabel("INTERFACE")
-      @JsonClass(generateAdapter = true)
+      @SerialName("INTERFACE")
+      @Serializable
       data class Interface(
           override val name: String,
           override val description: String?,
-          val kind: String,
```

Comment:
This is technicaly a breaking change but I think it's ok. I don't think many folks call the constructor outside the serialization.